apps/prairielearn/src/pages/course_instance/lookup/lookupByEnrollmentCode.ts

```diff
@@ -31,15 +35,55 @@ router.get(
 
     if (courseInstanceId && courseInstance.id !== courseInstanceId) {
       res.status(404).json({
-        error: 'No course found with this enrollment code',
+        error: 'This enrollment code is for a different course',
+      });
+      return;
+    }
+
+    const authnUser: User = res.locals.authn_user;
+
+    const existingEnrollment = await selectOptionalEnrollmentByUserId({
+      userId: res.locals.authn_user.user_id,
+      courseInstance,
+      requestedRole: 'Student', // TODO: Should be 'System'
+      authzData: dangerousFullAuthzForTesting(),
```

Comment:
This is only used for fancy error messages. we don't care about feeding it the correct auth because of that fact. Other places this is used follow the same pattern.